crates/red_knot_python_semantic/src/types/class_base.rs

```diff
@@ -224,3 +218,47 @@ impl<'db> From<&ClassBase<'db>> for Type<'db> {
         Self::from(*value)
     }
 }
+
+/// An iterator over the MRO of a class base.
+enum ClassBaseMroIterator<'db> {
+    Length2(core::array::IntoIter<ClassBase<'db>, 2>),
+    Length3(core::array::IntoIter<ClassBase<'db>, 3>),
+    UnknownLength(MroIterator<'db>),
```

Comment:
nit: I think `FromClass` would be a clearer name here (and `fn from_class` below)